### PR TITLE
add workflow to release

### PR DESCRIPTION
**Pull request by Vritra4**
3 binaries in total
- 2 binaries for linux/x86_64 and linux/aarch64
- 1 binary for darwin (universal)

## Summary by CodeRabbit

- **New Features**
	- Introduced automated workflows for building and releasing executable binaries for Darwin (macOS) and Linux platforms.
	- Added a release workflow that triggers builds based on version tags, streamlining the deployment process.

These enhancements ensure consistent and efficient builds for both operating systems, making it easier for users to access the latest versions of the software.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Build executable binaries and release them
+
+on:
+  push:
+    tags:
+      - 'v*'
+      - 'test-'
+
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/release.yml]: fixed

File: .github/workflows/build-linux.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+name: Build and realease Linux executable binaries
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-linux.yml]: fixed

File: .github/workflows/build-linux.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          rm build/opinitd
+
+      - name: Release
+        uses: softprops/action-gh-release@v2
+        with:
+          files: |
+            opinitd_${{ env.VERSION }}_Linux_x64_64.tar.gz
+            opinitd_${{ env.VERSION }}_Linux_aarch64.tar.gz
+        env:
+          VERSION: ${{ env.VERSION }}
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-linux.yml]: fixed

File: .github/workflows/build-darwin.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          go-version: "1.22.7"
+
+      - name: Set env vars
+        run: |
+          if [[ "${{ github.ref }}" == "refs/tags/"* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+          else
+            VERSION="v0.0.0-${GITHUB_SHA::8}"
+          fi
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-darwin.yml]: i'm not sure if it is mandatory but fixed anyway

**Review by LizJeong — APPROVED** [review]

File: .github/workflows/build-darwin.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          tar -czf "$GITHUB_WORKSPACE"/opinitd_"$VERSION"_Darwin.tar.gz ./build/opinitd
+          rm ./build/opinitd.arm64 ./build/opinitd.amd64 ./build/opinitd
+
+      - name: Release
+        uses: softprops/action-gh-release@v2
+        with:
+          files: |
+            opinitd_${{ env.VERSION }}_Darwin.tar.gz
+        env:
+          VERSION: ${{ env.VERSION }}
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-darwin.yml:45]: we are doing same action on the other repository without that option.

[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-darwin.yml:45]: @LizJeong any idea?

[the remark this answers is not in the document]
Comment (LizJeong) [reply .github/workflows/build-darwin.yml:45]: Tag conditions already exist at the workflow level - `release.yml`.
Therefore, adding them to individual job steps would be redundant.

[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-darwin.yml:45]: yeah let's resolve it

File: .github/workflows/build-linux.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          go-version: "1.22.7"
+
+      - name: Set env vars
+        run: |
+          if [[ "${{ github.ref }}" == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+          else
+            VERSION="v0.0.0-${GITHUB_SHA:0:8}"
+          fi
+          echo "VERSION=${VERSION}" >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-linux.yml]: i don't think it's necessary but updated it anyway...
cc: @LizJeong

[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-linux.yml]: @coderabiitai

[the remark this answers is not in the document]
Comment (Vritra4) [reply .github/workflows/build-linux.yml]: @coderabbitai

**Review by beer-1 — APPROVED** [review]
LGTM

**Pull request merged**